### DISCUSSION TITLE
Differentiate undefined and null in useRouteQuery

### DISCUSSION THREAD

**Discussion by some-user123** [vueuse/vueuse#2914] [Q&A]
I would like to differentiate these URLs:
1. `/foo`
1. `/foo?q`
1. `/foo?q=bar`

If I'm using `const route = useRoute()` that's possible;
1. `route.query.q === undefined`
1. `route.query.q === null`
1. `route.query.q === 'bar'`

But with `const q = useRouteQuery('q')` there seems no differentiation between `undefined` and `null`.
1. `q === null`
1. `q === null`
1. `q === 'bar'`

Is that intentional? Did I miss an obvious solution?

**Comment by matthew-dean** [comment]
I'm also confused by this. How do you set a querystring parameter without a value? `/foo?q` should be perfectly valid, and I would have also expected `null` to represent that value, and `undefined` to represent the parameter not being present.

**Reply by some-user123** [reply]
There is also an open issue on this topic: https://github.com/vueuse/vueuse/issues/3294

The PR to fix it, wasn't merged: https://github.com/vueuse/vueuse/pull/3304

**Comment by ilyaliao** [comment]
#3294 has been merged, so I think we can end the discussion here :)
